.circleci/config.yml

```diff
@@ -47,9 +45,20 @@ jobs:
                 name: install base64
                 command: apt update && apt install -y cl-base64
 
-            - run:
-                name: publish into Sonatype
-                command: .circleci/publish
+            - when:
+                condition:
+                  equal: [ develop, << pipeline.git.branch >> ]
+                steps:
+                  - run:
+                      name: publish into Sonatype
+                      command: .circleci/publish
+            - when:
+                condition:
+                  equal: [ master, << pipeline.git.branch >> ]
+                steps:
+                  - run:
+                      name: publish release
+                      command: .circleci/publishRelease
 
```

Comment:
It looks like we're building up quite a bit of redundancy: both `publish` and `publishRelease` have checks inside them to see they are on the right branch, and these have to be matched here by the conditional checks too. 

Wouldn't it be easier at this point to have just a single publish script that either:
* calls with `--release` or without depending on the branch, and then no conditional check is needed here, or
* keep the conditional and call `publish` or `publish --release` depending on which one matched, passing on the extra arguments?

Reply:
Maybe you are right here. I will got with `1` approach as i do not like `$@` as i am pretty sure everyone would need to always check what it means